specifyweb/workbench/upload/treerecord.py

```diff
@@ -388,135 +237,109 @@ def process_row_item(row_key: str, row_value: Any) -> List[TreeRankCell]:
             return tree_def_model.objects.get(id=target_rank_treedef_id)
 
-        # Retrieve the treedef items and root for the tree
-        def get_treedefitems_and_root(tree_rank_model, tree_node_model, target_rank_treedef_id: int):
-            # Fetch treedef items
-            def fetch_treedefitems():
-                return list(tree_rank_model.objects.filter(treedef_id=target_rank_treedef_id).order_by("rankid"))
-
-            # Fetch root node
-            def fetch_root():
-                return tree_node_model.objects.filter(definition_id=target_rank_treedef_id, parent=None).first()
+        tree_def_model, tree_rank_model, tree_node_model = get_models(self.name)
 
-            # Check if root is None and log warning
-            def check_root(root):
-                if root is None:
-                    logger.warning(f"No root found for treedef {target_rank_treedef_id}")
-                    return None, WorkBenchParseFailure('noRoot', {}, None)
-                return root
+        # Adjust scope if needed for old datasets
+        self = self._adjust_tree_scope()
 
-            treedefitems = fetch_treedefitems()
-            root = fetch_root()
-            root_checked = check_root(root)
+        ranks_columns_in_row_not_null = self._get_not_null_ranks_columns_in_row(row)
+        targeted_treedef_ids = get_targeted_treedefids(ranks_columns_in_row_not_null)
 
-            if root_checked is not root:
-                return root_checked
+        validation_result = self._run_validation_checks(targeted_treedef_ids, ranks_columns_in_row_not_null, row)
+        if validation_result: return validation_result
 
-            return treedefitems, root
+        target_rank_treedef_id = targeted_treedef_ids.pop()
+        target_rank_treedef = get_target_rank_treedef(tree_def_model, target_rank_treedef_id)
 
-        tree_def_model, tree_rank_model, tree_node_model = get_models(self.name)
+        # Based on the target treedef, get the treedefitems and root for the tree
+        treedefitems = list(tree_rank_model.objects.filter(treedef_id=target_rank_treedef_id).order_by("rankid"))
+        root = tree_node_model.objects.filter(definition_id=target_rank_treedef_id, parent=None).first()
 
+        return self._replace(treedef=target_rank_treedef, treedefitems=treedefitems, root=root), None
+    
+    """
+        Adjusts tree scope for TreeRankRecords with NULL treedef_ids.
+        Used mainly for legacy (pre-MOTs) datasets.
+        Newer datasets will have treeIds specified in the upload plan by the frontend
+    """
+    def _adjust_tree_scope(self):
+        adjusted_ranks = {}
+        for trr in self.ranks.keys():
+            adjusted_trr = trr
+            if trr.treedef_id is None:
+                adjusted_trr = trr._replace(treedef_id=self.treedef.id)
+                
+            adjusted_ranks[adjusted_trr] = self.ranks[trr]
+
+        return self._replace(ranks=adjusted_ranks)
```

Comment:
`TreeRankRecord` with null treedef_id will use the default treedef determined when scoping `TreeRecord`